vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -184,11 +184,10 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 	// DEPRECATED: remove and replace with something more accurate
 	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
 
-	if clientcmd.UseOpenShiftKubeConfigValues {
-		flags.StringVar(&loadingRules.ExplicitPath, clientcmd.OpenShiftKubeConfigFlagName, "", "Path to the kubeconfig file to use for CLI requests.")
-	} else {
-		flags.StringVar(&loadingRules.ExplicitPath, "kubeconfig", "", "Path to the kubeconfig file to use for CLI requests.")
-	}
+	flags.StringVar(&loadingRules.ExplicitPath, clientcmd.OpenShiftKubeConfigFlagName, "", "Path to the kubeconfig file to use for CLI requests.")
```

Comment:
This needs to be conditional.  When running as `kubectl` we don't want this flag available.

Reply:
Fixed.